internal/docs/design.md

```diff
@@ -107,6 +107,36 @@ symbols stable over time.
 The exception to this rule is if the name is not unique across providers. The
 canonical example is `gcpkms` and `awskms`.
 
+## Option Structs in APIs
+
+All public constructors should take an `Options` struct, even if it is currently
+empty, to ensure that we can add arguments to the APIs in the future without
+breaking backward compatibility.
+
+-   This includes driver constructors (e.g., `gcsblob.OpenBucket`) as well as
+    API functions (e.g., `blob.NewReader`). When in doubt, if you think it's
+    possible that we'll add arguments, add `Options`.
+-   Name the `Options` struct appropriately. `Options` is usually fine for
+    provider constructors since the package generally only exposes a
+    constructor. Inside a driver interface or in a concrete type like `blob`,
+    use more descriptive names like `ReaderOptions` or `WriterOptions`.
+-   When similar `Options` are part of a driver interface and also part of the
+    concrete type (e.g., `blob.WriterOptions`), duplicate the struct instead of
+    aliasing or embedding it, and copy the struct fields explicitly where
+    needed. This allows the godoc for each type to be tailored to the
+    appropriate audience (concrete type: users, driver interface: provider
+    implementors), and also allows the structs to diverge over time if
+    appropriate.
+-   Arguments that are optional, including ones for which a default value is
+    common, should generally go into the `Options` struct. Avoid adding
+    `Options` fields for which the default value doesn't make sense.
+-   Regarding empty `Options` structs: we considered only adding them the first
```

Comment:
This prose feels like it should be part of the first paragraph when it is introducing the idea. The rest of the bullets work nicely to explain the guidelines efficiently.

Reply:
It's a pretty long aside, and I think distracts from the main points. I pulled it out of the bullet list and left it at the end.